api/include/opentelemetry/trace/propagation/composite_propagator.h

```diff
@@ -0,0 +1,73 @@
+#include <initializer_list>
+#include <memory>
+#include <vector>
+#include "opentelemetry/trace/propagation/text_map_propagator.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+namespace propagation
+{
+
+template <typename T>
+class CompositePropagator : public TextMapPropagator<T>
+{
+public:
+  CompositePropagator(std::vector<std::shared_ptr<TextMapPropagator<T>>> &propagators)
```

Comment:
A few minor nitpicks:
* Should `CompositePropagator` own the propagators? i.e. use unique_ptr over shared_ptr?
* Pass by value and `std::move`?
* Would it help to add a method to add a propagator to composite propagator?

Reply:
> * Should `CompositePropagator` own the propagators? i.e. use unique_ptr over shared_ptr?
> * Pass by value and `std::move`?

Very valid points. I can't think of scenario where we need to share Propagator across. Will fix this, unless someone further comments with some scenario.

> * Would it help to add a method to add a propagator to composite propagator?

This should help if we need to add propagator after creating the initial composite object. Specs ( https://github.com/open-telemetry/opentelemetry-specification/blob/14d123c121b6caa53bffd011292c42a181c9ca26/specification/context/api-propagators.md#composite-propagator ) doesn't ask for it explicitly so would keep it as it for now. We can add later if required.

Reply:
First two points are implemented now.